deps/rabbitmq_peer_discovery_common/src/rabbit_peer_discovery_cleanup.erl

```diff
@@ -241,7 +243,7 @@ maybe_cleanup(State, UnreachableNodes) ->
        "Peer discovery: cleanup discovered unreachable nodes: ~tp",
        [UnreachableNodes],
        #{domain => ?RMQLOG_DOMAIN_PEER_DIS}),
-    case lists:subtract(UnreachableNodes, service_discovery_nodes()) of
+    case lists:subtract(as_list(UnreachableNodes), as_list(service_discovery_nodes())) of
```

Comment:
This can't work with a single seed node. The cleanup should be skipped because we don't have the list of members.

If we use the seed node as a list, this code will uncluster everyone.